agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/processors/AgentProcessor.java

```diff
@@ -85,15 +90,21 @@ public boolean isMatch(Attributes attributes, String name) {
     private boolean checkAttributes(Attributes attributes) {
       for (ProcessorAttribute attribute : processorAttributes) {
         // All of these attributes must match exactly for a match to occur.
-        Object existingAttributeValue = attributes.get(AttributeKey.stringKey(attribute.key));
-        // to get the string value
-        // existingAttributeValue.toString()
-        // String.valueOf(existingAttributeValue);
-        if (!(existingAttributeValue instanceof String)) {
+        Object valueObject = attributes.get(attribute.getAttributeKey());
+        if (valueObject == null) {
           // user specified key not found
           return false;
         }
-        if (attribute.value != null && !existingAttributeValue.equals(attribute.value)) {
+        String existingAttributeValue;
+        if (attribute.type == BOOLEAN_ARRAY
+            || attribute.type == LONG_ARRAY
+            || attribute.type == DOUBLE_ARRAY
+            || attribute.type == STRING_ARRAY) {
+          existingAttributeValue = valueObject.toString();
```

Comment:
could you please add a comment explaining why `String.valueOf` would not work in this case

Reply:
no diff. updated.